metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceImpl.java

```diff
@@ -323,93 +319,108 @@ public void notifyOfTableRename(final MetacatRenameTablePostEvent event) {
+        @Nullable final String fallbackArn,
         final SNSMessage<?> message,
         final QualifiedName name,
         final String errorMessage,
-        final String counterKey,
-        final boolean retryOnLongMessage
+        final String counterKey
     ) {
         this.notificationMetric.recordTime(message, Metrics.TimerNotificationsBeforePublishDelay.getMetricName());
         try {
-            final AsyncHandler<PublishRequest, PublishResult> handler =
-                new AsyncHandler<PublishRequest, PublishResult>() {
-                    @Override
-                    public void onError(final Exception exception) {
-                        if (retryOnLongMessage && (exception instanceof InvalidParameterException
-                            || exception instanceof InvalidParameterValueException)) {
-                            log.error("SNS Publish message exceeded the size threshold", exception);
-                            notificationMetric.counterIncrement(
-                                Metrics.CounterSNSNotificationPublishMessageSizeExceeded.getMetricName());
-                            final SNSMessage<Void> voidMessage = new SNSMessage<>(message.getId(),
-                                message.getTimestamp(), message.getRequestId(), message.getType(), message.getName(),
-                                null);
-                            publishNotification(arn, voidMessage, name, errorMessage, counterKey, false);
-                        } else {
-                            notificationMetric.handleException(name, errorMessage, counterKey, message, exception);
-                        }
-                    }
-
-                    @Override
-                    public void onSuccess(final PublishRequest request, final PublishResult publishResult) {
-                        log.info("Successfully published message to topic {} with id {}",
-                            arn, publishResult.getMessageId());
-                        log.debug("Successfully published message {} to topic {} with id {}",
-                            message, arn, publishResult.getMessageId());
-                        notificationMetric.counterIncrement(counterKey);
-                        notificationMetric.recordTime(message,
-                            Metrics.TimerNotificationsPublishDelay.getMetricName());
-                    }
-                };
-            client.publishAsync(arn, mapper.writeValueAsString(message), handler);
-        } catch (final Exception e) {
+            //
+            // Publish the event to original SNS topic. If we receive an error from SNS, we will then try publishing
+            // to the fallback topic.
+            //
+            try {
+                publishNotification(arn, message, counterKey);
+            } catch (final Exception exception) {
+                if (fallbackArn != null) {
+                    log.info("Fallback published message to topic {} because of error {}",
```

Comment:
Are we assuming microbot listening on the fallback SNS as well?

Reply:
Also, how can we order the messages between these two SNS? For example, the events of the same table ended up into two SNS, how the consumer know which order to process them?

Reply:
Sounds good to me.